src/lib/routes/controller.ts

```diff
@@ -105,6 +112,44 @@ export default class Controller {
         );
     }
 
+    routeWithOpenApi(openApiService: OpenApiService) {
+        return ({
+            openApi,
+            ...options
+        }: IRouteOptions & { openApi: ApiOperation }): void => {
+            const errorCodes = new Set<StandardResponseCodes>([401]);
+
+            if (
+                ['put', 'post', 'patch'].includes(
+                    options?.method?.toLowerCase() || '',
+                )
+            ) {
+                errorCodes.add(400);
```

Comment:
we can also analyze the path for path params and generate 404 automatically

Reply:
Yeah, I thought about that! The problem is that we don't always return 404s if the path param doesn't match anything, though. But I agree that we should, ideally. However, I wouldn't necessarily be that against adding it. What do you think?

Reply:
Sure! I think that's primarily for our DELETEs, though, so we could do a check on that too, I suppose? If it has path params and isn't a DELETE, then ...

Reply:
Coming back to this: I just read up a bit on it, and I think that deletes actually **should** return 404 if the resource doesn't exist. So I'd be happy to mark that as the intention and we could maybe strive towards making that a reality.

Reply:
Realized right now that the 404 won't apply correctly to the token name endpoint you created yesterday, if it returns an empty list when no tokens are found. Would you call that an exception to the rule, or should it actually be a 404? 🤔 I'm kinda torn, to be honest. I can see both being valid options.